src/proposal.ts

```diff
@@ -511,25 +511,46 @@ export class Proposal implements IStateful<IProposalState> {
       const threshold = realMathToNumber(new BN(item.gpQueue.threshold))
       const stakesFor = new BN(item.stakesFor)
       const stakesAgainst = new BN(item.stakesAgainst)
+      /**
+       * for doing multiplication between floating point (threshold) and BN numbers
+       */
+      const PRECISION = Math.pow(2, 40)
 
-      // upstakeNeededToPreBoost is the amount of tokens needed to upstake to move to the preboost queue
-      // this is only non-zero for Queued proposals
-      // note that the number can be negative!
+      /**
+       * The number of up-staking tokens (usually GEN) needed to qualify a queued proposal to move into the
+       * pre-boosted queue.
+       *
+       * Only computed for queued proposals.
+       *
+       * The equation is derived from: threshold = (stakesFor + upstakeNeededToPreBoost) / stakesAgainst
+       *
+       * >=  0 : any number of up-staking tokens greater than this will qualify to move the proposal
+       *        to the preboost queue
+       * <  0 : the proposal ought already to be pre-boosted
+       */
       let upstakeNeededToPreBoost: BN = new BN(0)
-      const PRECISION = Math.pow(2, 40)
       if (stage === IProposalStage.Queued) {
-
         upstakeNeededToPreBoost = new BN(threshold * PRECISION)
           .mul(stakesAgainst)
           .div(new BN(PRECISION))
           .sub(stakesFor)
       }
-      // upstakeNeededToPreBoost is the amount of tokens needed to upstake to move to the Queued queue
-      // this is only non-zero for Preboosted proposals
-      // note that the number can be negative!
+
+      /**
+       * The number of down-staking tokens (usually GEN) needed to qualify a pre-boosted proposal to move back
+       * to the Queued queue.
+       * Only computed for PreBoosted proposals.
+       *
+       * The equation is derived from: threshold = stakesFor / (stakesAgainst + downStakeNeededToQueue)
+       *
+       * >  0 : any number of down-staking tokens greater-than-or-equal to this will qualify to move the proposal
```

Comment:
same comment as above :-)

Reply:
I clarified the comments.  I think they provide more info than is readily available by thinking about the equations.